pkg/compact/tsdb_adapter.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package compact
+
+import (
+	"github.com/oklog/ulid"
+	"github.com/prometheus/prometheus/tsdb"
+)
+
+type tsdbCompactorAdapter struct {
+	compactor *tsdb.LeveledCompactor
+}
+
+func NewTSDBCompactorAdapter(compactor *tsdb.LeveledCompactor) *tsdbCompactorAdapter {
+	return &tsdbCompactorAdapter{compactor: compactor}
+}
+
+func (a tsdbCompactorAdapter) Compact(dest string, dirs []string, open []*tsdb.Block) ([]ulid.ULID, error) {
+	uid, err := a.compactor.Compact(dest, dirs, open)
```

Comment:
It looks like this function can return an empty ULID. Should we check for that and return an empty slice in that case?

Reply:
No. With the interface change, we ensure all returned ULIDs are not empty

Reply:
Isn't `tsdb.LeveledCompactor` also returns a slice rather than a single ULID? https://github.com/prometheus/prometheus/blob/main/tsdb/compact.go#L448

Reply:
Maybe we should wait and upgrade to 0.54?

Reply:
Yeah that makes more sense to me. I am fine of using main as well